api/v1alpha1/shared_types.go

```diff
@@ -191,6 +191,13 @@ type KubernetesServiceSpec struct {
 	// +optional
 	AllocateLoadBalancerNodePorts *bool `json:"allocateLoadBalancerNodePorts,omitempty"`
 
+	// LoadBalancerIP defines the IP Address of the underlying load balancer service. This field
+	// may be ignored if the load balancer provider does not support this feature.
+	// This field has been Deprecated, but it is still used for set the IP Address in some cloud
+	// providers such as GCP.
```

Comment:
it's deprecated, maybe use annotations to support this?

Reply:
Not possible for GCP, the linked issue has all the details:)

Reply:
thanks for clearify.

Reply:
what will happen if someone config this by mistake on non-GCP?

Reply:
https://github.com/envoyproxy/gateway/issues/1841#issuecomment-1699472763
has the details, looks like all still support it except AWS and I'm unsure what happens when it is set, any idea @LanceEa ?

Reply:
@zirain - see discussion here: https://github.com/envoyproxy/gateway/issues/1841#issuecomment-1699472763

tl;dr; using term "deprecated" is probably not correct but rather discouraged until k8s community/cloud providers have completely moved away from it.

Reply:
Just want to make sure, it won't break anything.

Reply:
Thanks @LanceEa can you please review the API doc string in this PR

Reply:
My guess is either it is ignored or like GCP the admission controller would reject the fields it doesn't support.

Reply:
What I'm thinking is should EG auto-detect cloud provider, and reject/ignore this field if it's non-GCP?

Reply:
@arkodg - I agree. At that point I think we would be over engineering  it and would end up with more pain maintaining it. Instead I think leaning on documentation is a better approach.